warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -2535,14 +2543,22 @@ public static GTSEncoder parse(GTSEncoder encoder, String str, Map<String,String
     idx2 = UnsafeString.indexOf(str, ' ', idx);
     
     if (-1 == idx2){
-      throw new ParseException(str, idx);
+      if(0 == tsoffset) {
+        throw new ParseException("Missing GTS name, labels and value.", idx);
```

Comment:
The actual incorrect line is no longer shown.

Reply:
Ingress and its declination already include the offending line when catching the ParseException.

I think it's a good way of doing things: the ParseException only tells where on the line there is a problem and what is the problem and the caller knows which lines caused the problem, so it's up to the caller to include the line in the Exception or error message.

I updated the error messages to include the offset where is was not and better handled the errors in PARSE.

Reply:
GTSHelper#parse is used in PARSE also where the offending line is not reported in case of error, this could be a nice addition to this PR. Same with StreamingUpdate endpoints if that is not already the case.

Reply:
That's exactly the changes in a433a5d.

Reply:
My bad, sorry.